InscryptionAPI/Triggers/CustomTriggerPatches.cs

```diff
@@ -6,209 +6,281 @@
         [HarmonyPostfix]
-        private static IEnumerator TriggerDeathTriggers(IEnumerator result, PlayableCard __instance, bool wasSacrifice, PlayableCard killer = null)
+        public static IEnumerator TriggerDeathTriggers(IEnumerator result, PlayableCard __instance, bool wasSacrifice, PlayableCard killer = null)
         {
             CardSlot slotBeforeDeath = __instance.Slot;
             while (result.MoveNext())
             {
                 yield return result.Current;
-                if (result.Current.GetType() == TriggerCardsOnBoardEnumerator)
+                if (result.Current.GetType() == triggerType)
                 {
-                    Trigger t = (Trigger)TriggerField.GetValue(result.Current);
+                    Trigger t = Trigger.None;
+                    try
+                    {
+                        t = (Trigger)result.Current.GetType().GetField("trigger").GetValue(result.Current);
+                    }
+                    catch { }
                     if (t == Trigger.OtherCardPreDeath)
                     {
-                        foreach (var pre in CustomTriggerFinder.FindTriggersInHand<IOtherCardPreDeathInHand>())
-                        {
-                            if (pre.RespondsToOtherCardPreDeathInHand(slotBeforeDeath, wasSacrifice, killer))
-                                yield return pre.OnOtherCardPreDeathInHand(slotBeforeDeath, wasSacrifice, killer);
-                        }
+                        yield return CustomGlobalTriggerHandler.TriggerInHand<IOnOtherCardPreDeathInHand>(x => x.RespondsToOtherCardPreDeathInHand(slotBeforeDeath, !wasSacrifice, killer), x =>
+                            x.OnOtherCardPreDeathInHand(slotBeforeDeath, !wasSacrifice, killer));
                     }
                     else if (t == Trigger.OtherCardDie)
                     {
-                        foreach (var i in CustomTriggerFinder.FindTriggersInHand<IOtherCardDieInHand>())
+                        yield return CustomGlobalTriggerHandler.TriggerInHand<IOnOtherCardDieInHand>(x => x.RespondsToOtherCardDieInHand(__instance, slotBeforeDeath, !wasSacrifice, killer), x =>
+                            x.OnOtherCardDieInHand(__instance, slotBeforeDeath, !wasSacrifice, killer));
+                    }
+                }
+            }
+            yield break;
+        }
+
+        [HarmonyPatch(typeof(PlayableCard), nameof(PlayableCard.TakeDamage))]
+        [HarmonyPostfix]
+        public static IEnumerator TriggerOnTurnEndInHandPlayer(IEnumerator result, PlayableCard __instance, int damage, PlayableCard attacker)
+        {
+            CustomGlobalTriggerHandler.CollectDataAll<ICardTakenDamageModifier, int>(true, x => x.RespondsToCardTakenDamageModifier(__instance, damage), x => damage = x.CollectCardTakenDamageModifier(__instance, damage));
+            if (damage != 0)
+            {
+                result?.GetType()?.GetField("damage")?.SetValue(result, damage);
```

Comment:
Only commenting here but occurs in multiple places, please cache as much reflection as possible

Reply:
is there any other way to change values of an ienumerator?